query/src/test/scala/filodb/query/PromCirceSupportSpec.scala

```diff
@@ -0,0 +1,98 @@
+package filodb.query
+
+import io.circe.parser
+import io.circe.generic.auto._
+//import io.circe.syntax.EncoderOps
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+
+class PromCirceSupportSpec extends AnyFunSpec with Matchers with ScalaFutures {
+
+  // DO NOT REMOVE PromCirceSupport import below assuming it is unused - Intellij removes it in auto-imports :( .
+  // Needed to override Sampl case class Encoder.
+  import PromCirceSupport._
+
+  it("should parse Sampl") {
+    val inputString =
+      """
+        |[
+        |	[
+        |		1600102672,
+        |		"1.2"
+        |	],
+        |	[
+        |		1600102687,
+        |		"3.1"
+        |	]
+        |]
+        |""".stripMargin
+
+   parseAndValidate(inputString, List(Sampl(1600102672,1.2), Sampl(1600102687,3.1)))
+  }
+
+  it("should parse aggregateResponse") {
+    val input = """[{
+                  |	"status": "success",
+                  |	"data": {
+                  |		"resultType": "matrix",
+                  |		"result": [{
+                  |			"metric": {
+                  |
+                  |			},
+                  |			"aggregateResponse": {
+                  |				"aggregateValues": [
+                  |					[
+                  |						1601491649,
+                  |						"15.186417982460787",
+                  |						5
```

Comment:
How will NaN be encoded ?

Reply:
It will be a string "NaN". I have added it to test case